File: DotnetAsm.Frontend/src/models/AsmGenerationRequest.ts
import TargetFramework from "./TargetFramework";

export default class AsmGenerationRequest {
  csharpCode!: string;
  methodName?: string;
  generateSummary!: boolean;
  usePgo!: boolean;
  useTieredCompilation!: boolean;
  useReadyToRun!: boolean;
  targetFramework!: TargetFramework;

  constructor(
    csharpCode: string,
    methodName: string,
    generateSummary: boolean,
    usePgo: boolean,
    useTieredCompilation: boolean,
    useReadyToRun: boolean,
    targetFramework: TargetFramework
  ) {
    this.csharpCode = csharpCode;
    this.methodName = methodName;
    this.generateSummary = generateSummary;
    this.usePgo = usePgo;
    this.useTieredCompilation = useTieredCompilation;
    this.useReadyToRun = useReadyToRun;
    this.targetFramework = targetFramework;
  }
}

File: DotnetAsm.Core/Models/AsmGenerationRequest.cs
namespace DotnetAsm.Core.Models;

public class AsmGenerationRequest
{
    public required string CsharpCode { get; init; }
    public string? MethodName { get; init; }
    public required bool UsePgo { get; init; }
    public required bool UseTieredCompilation { get; init; }
    public required bool UseReadyToRun { get; init; }
    public TargetFramework TargetFramework { get; init; }
}